staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -327,7 +312,25 @@ func newAuthenticator(opts Options, initVerifier func(ctx context.Context, a *Au
 		resolver:       resolver,
 	}
 
-	initVerifier(ctx, authenticator, verifierConfig)
+	if opts.KeySet != nil {
+		// We already have a key set, synchronously initialize the verifier.
+		authenticator.setVerifier(oidc.NewVerifier(opts.IssuerURL, opts.KeySet, verifierConfig))
```

Comment:
Do we care to dynamically reload the `KeySet` like we do in the `else` block? This might be helpful for someone who wants to provide an out-of-band rotating set of keys dynamically.

Reply:
`KeySet` is an interface, not a struct.  Users are free to rotate the keys as they see fit.  This is what the default `remoteKeySet` does automatically via the `jwks_uri` data.